src/TextProviderHook.js

```diff
@@ -0,0 +1,14 @@
+import { useContext } from 'react';
+import { TextContext } from './TextProvider';
+
+function useTextProvider(context = TextContext) {
+  const globalText = useContext(context);
+  return (key) => {
+    if (Object.prototype.hasOwnProperty.call(globalText, key)) {
```

Comment:
It will be better to copy the function from FormattedMessage.jsx rather than the HOC as that has support for values also.

Reply:
It would add consistent behaviour. In that process if you can fix the HOC component also that would be great.

Reply:
Let me know if the most recent commits are what you're looking for.